features/device-config/sink/consumer/src/main/java/org/opennms/features/deviceconfig/sink/consumer/DeviceConfigConsumer.java

```diff
@@ -106,6 +108,19 @@ public void handleMessage(DeviceConfigSinkDTO message) {
         }
     }
 
+    private OnmsIpInterface findMatchingInterface(final String ipAddress, final String location) {
```

Comment:
Perhaps make `DeviceConfigService.findMatchingInterface` public, then call it here (with `serviceName` = `null`) rather than duplicate code? Not sure if that introduces any dependency issues.